### ISSUE TITLE
Is it possible to start Foreground initiated jobs that continue background execution?

### ISSUE THREAD

**Issue by kelvinharron** [lucas34/SwiftQueue#405]
Hi folks, hats off to SwiftQueue as an impressive and easy to use piece of work! I have a question regarding its capabilities concerning what I need our team to achieve. I ask this because I have a very simple spike that suggests the restrictions on iOS mean it may not be achievable. 

### Goal

When the user completes a behaviour in our app, we need to complete a few work steps, including an API request and a video upload as one Job. We need to queue this immediately and execute it if the user has connectivity, and if not, then attempt the upload on the subsequent wake of the app. For users with no connectivity, there could be several jobs of this behaviour to execute and the time to execute will vary based on video size.

I'm aware of the BackgroundTasks API (that is backing some functionality in SwiftQueue), but my understanding is that Apple would prefer you do very little to no work in the background (with little time permitted by default).

### SwiftQueue spike implementation

By following your well-documented API, I've created a simple spike project (attached below) that uses the described jobBuilder, and in my test scenario, I simply click the button to invoke `submit()` which launches a task with a sleep for three seconds where I background the app, hoping for the app to print success. For me, this only occurs when resuming foreground.

```
private let jobBuilder = JobBuilder(type: UploadJob.type)
private let swiftQueueManager = SwiftQueueManagerBuilder(creator: UploadWorkerBuilder()).build()

...

func submit() {
    jobBuilder
            .internet(atLeast: .cellular)
            .service(quality: .userInitiated) // attempting to highlight to the system its an important piece of work
            .with(params: ["data": uploadCounter])
            .retry(limit: .limited(5))
            .persist()
            .periodic(limit: .limited(1), executor: .any) // any executor for both foreground and background
            .schedule(manager: swiftQueueManager)
}
```

[SwiftQueueSpike.zip](https://github.com/lucas34/SwiftQueue/files/8898612/SwiftQueueSpike.zip)

### The Question

To put it simply, is it possible using SwiftQueue to start a foreground job, and once the app is backgrounded (not so worried about being killed), it can continue to execute? While reading some of the previously closed issues here, it does not seem to be feasible, but I'd appreciate an insight, if available, given your exposure to the Background Tasks API.

Many thanks in advance!

**Comment by lucas34** [comment]
Hi, it sounds like an iOS problem. Not sure how SwiftQueue can help on this
